Code/autopkglib/FlatPkgUnpacker.py

```diff
@@ -111,17 +111,14 @@ def xar_expand(self):
             ]
             if self.env.get("skip_payload"):
                 xarcmd.extend(["--exclude", "Payload"])
-            proc = subprocess.Popen(
-                xarcmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True
-            )
-            (_, stderr) = proc.communicate()
+            proc = subprocess.run(xarcmd, capture_output=True, text=True)
         except OSError as err:
             raise ProcessorError(
                 f"xar execution failed with error code {err.errno}: {err.strerror}"
             )
         if proc.returncode != 0:
             raise ProcessorError(
-                f"extraction of {self.env['flat_pkg_path']} with xar failed: {stderr}"
+                f"extraction of {self.env['flat_pkg_path']} with xar failed: {proc.stderr}"
             )
```

Comment:
Same feedback here as above - we should leverage check=True.

Reply:
3f8acae